src/http2_connection.erl

```diff
@@ -987,17 +1004,19 @@ handle_event({send_headers, StreamId, Headers},
      Conn#connection{
        encode_context=NewContext
       }};
-handle_event({send_body, StreamId, Body},
+handle_event({send_body, StreamId, Body, Opts},
              StateName,
              #connection{}=Conn) ->
     lager:debug("[~p] Send Body Stream ~p",
                 [Conn#connection.type, StreamId]),
     Stream = get_stream(StreamId, Conn#connection.streams),
+    BodyComplete = proplists:get_value(send_end_stream, Opts, true),
     {NewSWS, NewS} =
         s_send_what_we_can(Conn#connection.send_window_size,
                            Conn#connection.send_settings#settings.max_frame_size,
                            Stream#stream{
-                            queued_data=Body
+                            queued_data=Body,
```

Comment:
Need to change this line to something along the lines of

```erlang
queued_data=<<Stream#stream.queued_data,Body>>,
```

but really it needs to be safer than that, something like

```erlang
NewBody = case is_binary(Stream#stream.queued_data) of
    true ->
        <<Stream#stream.queued_data,Body>>;
    false ->
        Body
end,
{NewSWS, NewS} =
   s_send_what_we_can(Conn#connection.send_window_size,
                      Conn#connection.send_settings#settings.max_frame_size,
                      Stream#stream{
                          queued_data=NewBody,
```

That and a supporting test and this PR should be ready to merge